perceval/backends/_slos.py

```diff
@@ -197,7 +198,7 @@ def prob_distribution(self) -> BSDistribution:
         c = abs(c) ** 2 / istate.prodnfact()
         xq.all_prob_normalize_output(c, self._fsas[istate.n])
         bsd = BSDistribution()
-        for output_state, probability in zip(allstate_iterator(self._input_state, self._mask), c):
+        for output_state, probability in zip(self._get_iterator(self._input_state, self._mask), c):
```

Comment:
would this change be only in _slos?

Reply:
Nope. The _get_iterator method comes from the AProbampliBackend class and can be used in any backend